File: pkg/utils/shell.go
package utils

import (
	"fmt"
	"os"
	"os/exec"
)

// RunShellCommand runs a shell command and streams its output to the terminal
func RunShellCommand(name string, args ...string) error {
	cmd := exec.Command(name, args...)
	cmd.Stdout = os.Stdout
	cmd.Stderr = os.Stderr
	cmd.Stdin = os.Stdin

	fmt.Printf("\n▶️ Running command: %s %v\n", name, args)
	return cmd.Run()
}

File: src/utils/shell.ts
import { execa } from 'execa';

export interface ShellResult {
  stdout: string;
  stderr: string;
  exitCode: number;
  success: boolean;
}

export async function runCommand(command: string, args: string[] = [], options: { cwd?: string } = {}): Promise<ShellResult> {
  try {
    const result = await execa(command, args, {
      cwd: options.cwd || process.cwd(),
      shell: true,
      reject: false,
    });

    return {
      stdout: result.stdout,
      stderr: result.stderr,
      exitCode: result.exitCode || 0,
      success: result.exitCode === 0,
    };
  } catch (error: any) {
    return {
      stdout: '',
      stderr: error.message || String(error),
      exitCode: 1,
      success: false,
    };
  }
}

export async function runShellCommand(command: string, options: { cwd?: string } = {}): Promise<ShellResult> {
  try {
    const result = await execa(command, {
      cwd: options.cwd || process.cwd(),
      shell: true,
      reject: false,
    });

    return {
      stdout: result.stdout,
      stderr: result.stderr,
      exitCode: result.exitCode || 0,
      success: result.exitCode === 0,
    };
  } catch (error: any) {
    return {
      stdout: '',
      stderr: error.message || String(error),
      exitCode: 1,
      success: false,
    };
  }
}
